tests/foreman/cli/test_flatpak.py

```diff
@@ -338,3 +353,154 @@ def test_sync_consume_flatpak_repo_via_library(
+    request,
+    module_target_sat,
+    module_flatpak_contenthost,
+    function_host_cleanup,
+    function_org,
+    function_lce,
+    function_product,
+    function_flatpak_remote,
+):
+    """Verify flatpak repository workflow end to end via CV.
+
+    :id: 820b59b0-20f0-472a-afab-5c480bedbfd0
+
+    :parametrized: yes
+
+    :setup:
+        1. Create a Product and a Lifecycle Environment.
+        2. Create a flatpak remote and scan it.
+
+    :steps:
+        1. Mirror flatpak repositories and sync them.
+        2. Create two CVs, put different repos inside, publish and promote them to LCE.
+        3. Create an AK assigned with one CVE only.
+        4. Register a content host using the AK.
+        5. Configure the content host to use Satellite's flatpak index.
+        6. Ensure only the proper Apps are available.
+        7. Install flatpak app from the first CV, ensure it succeeded.
+        8. Try to install flatpak app from the second CV, ensure it failed.
+
+    :expectedresults:
+        1. Flatpak repos published in a CV are installable on a host via the CV.
+        2. Other flatpak repos published in a different CV are isolated from the first CV.
+
+    """
+    sat, host = module_target_sat, module_flatpak_contenthost
+
+    # 1. Mirror flatpak repositories and sync them.
+    repo_names = ['rhel9/firefox-flatpak', 'rhel9/inkscape-flatpak', 'rhel9/flatpak-runtime']
+    remote_repos = [r for r in function_flatpak_remote.repos if r['name'] in repo_names]
+    for repo in remote_repos:
+        sat.cli.FlatpakRemote().repository_mirror(
+            {
+                'flatpak-remote-id': function_flatpak_remote.remote['id'],
+                'id': repo['id'],
+                'product-id': function_product.id,
+            }
+        )
+        local_repo = sat.cli.Repository.list(
+            {'product-id': function_product.id, 'name': repo['name']}
+        )[0]
+        sat.cli.Repository.synchronize({'id': local_repo['id']})
+
+    local_repos = sat.cli.Repository.list({'product-id': function_product.id})
+    assert set([r['name'] for r in local_repos]) == set(repo_names), (
+        'Required repo(s) were not scanned or mirrored'
+    )
+
+    # 2. Create two CVs, put different repos inside, publish and promote them to LCE.
+    cv1 = sat.api.ContentView(
+        organization=function_org,
+        repository=[
+            r['id']
+            for r in local_repos
+            if r['name'] in ['rhel9/inkscape-flatpak', 'rhel9/flatpak-runtime']
+        ],
+    ).create()
+    cv2 = sat.api.ContentView(
+        organization=function_org,
+        repository=[
+            r['id']
+            for r in local_repos
+            if r['name'] in ['rhel9/firefox-flatpak', 'rhel9/flatpak-runtime']
+        ],
+    ).create()
+    for cv in [cv1, cv2]:
+        cv.publish()
+        cv.read().version[0].promote(data={'environment_ids': function_lce.id})
+
+    # 3. Create an AK assigned with one CVE only.
```

Comment:
```suggestion
    # 3. Create an AK assigned with one CV only.
```
I'm guessing you meant convent view here, but feel free to ignore if you did mean CVE.

Reply:
I'm trying _really_ hard not to get in the habit of using "CVE" as an abbreviation for _content view environment_. It may be a losing battle..

Reply:
Hehe, I got the impression that "CVE" is the new fancy way to name "particular CV version promoted to particular LCE" (and it seems so nicely short and descriptive to me)

But I can step back and update the PR with something like "CV/LCE" not to spread that nice habit further. :smiling_face_with_tear:

Reply:
@vsedmik I don't want to be too prescriptive here, but for me, using `CVE` to mean anything other than `Common Vulnerabilities and Exposures` in this context is confusing enough that it actually reduces code readability, because it's such an industry-standard abbreviation. It would be like coming up with a new meaning for `TCP`.

This is ultimately your decision, and if using CVE to mean `content view environment` is already an established practice, then I acknowledge that there's not much I can do about that. But if the habit is not fully established, then I would encourage both you and @jeremylenz to keep up the fight.

Reply:
yeah, I'm trying _not_ to establish it :) Usually I try to type out "content view environment" fully. Or sometimes I just use "environment," but that's not ideal either because we already have about 3 meanings for that word.

Reply:
Oh okay, finally got your point with the `Common Vulnerabilities and Exposures` parallel. Captions updated.